src/pages/settings/Wallet/WalletPage/WalletPage.js

```diff
@@ -352,23 +358,44 @@ function WalletPage({bankAccountList, betas, cardList, fundList, isLoadingPaymen
                                                 </OfflineWithFeedback>
                                             )}
                                             <KYCWall
-                                                onSuccessfulKYC={navigateToTransferBalancePage}
+                                                onSuccessfulKYC={(_iouPaymentType, source) => navigateToWalletOrTransferBalancePage(source)}
+                                                onSelectPaymentMethod={(selectedPaymentMethod) => {
+                                                    if (hasActivatedWallet || selectedPaymentMethod !== CONST.PAYMENT_METHODS.BANK_ACCOUNT) {
+                                                        return;
+                                                    }
+                                                    // To allow upgrading to a gold wallet, continue with the KYC flow after adding a bank account
+                                                    BankAccounts.setPersonalBankAccountContinueKYCOnSuccess(true);
+                                                }}
                                                 enablePaymentsRoute={ROUTES.SETTINGS_ENABLE_PAYMENTS}
                                                 addBankAccountRoute={ROUTES.SETTINGS_ADD_BANK_ACCOUNT}
                                                 addDebitCardRoute={ROUTES.SETTINGS_ADD_DEBIT_CARD}
                                                 popoverPlacement="bottom"
+                                                source={hasActivatedWallet ? CONST.KYC_WALL_SOURCE.TRANSFER_BALANCE : CONST.KYC_WALL_SOURCE.ENABLE_WALLET}
```

Comment:
If the wallet is already activated, we won't trigger the KYCWall at all, right? Would we need to set the source here still?

Reply:
> I'm not sure if the wallet is deactivated automatically in that case

With the new change to [hasExpensifyPaymentMethod](https://github.com/Expensify/App/pull/29915), if the bank account is removed, hasExpensifyPaymentMethod will return false so we should be good here.

Thanks for clarifying, I think this logic makes sense 👍🏼